src/Runners/PHPUnit/Options.php

```diff
@@ -313,7 +335,41 @@ public static function fromConsoleInput(InputInterface $input, string $cwd): sel
             throw new InvalidArgumentException('Option --filter is not implemented for non functional mode');
         }
 
+        if (isset($options['order-by']) && ! in_array($options['order-by'], self::ORDER_TYPES, true)) {
+            throw new InvalidArgumentException('Option --order-by supports only ' . implode('|', self::ORDER_TYPES));
+        }
+
+        if (isset($options['random-order-seed'])) {
+            if (! is_numeric($options['random-order-seed'])) {
+                throw new InvalidArgumentException(sprintf(
+                    'Option --random-order-seed should have a number value, "%s" given',
+                    (string) $options['random-order-seed']
+                ));
+            }
+
+            if (! isset($options['order-by'])) {
+                throw new InvalidArgumentException('Option --random-order-seed useless without --order-by=random');
+            }
+
+            if ($options['order-by'] !== self::ORDER_RANDOM) {
+                throw new InvalidArgumentException(sprintf('Option --random-order-seed useless in order-by=%s mode', $options['order-by']));
+            }
+        }
+
         $filtered = [];
+
+        if (isset($options['order-by'])) {
+            $filtered['order-by'] = $options['order-by'];
+
+            if ($options['order-by'] === self::ORDER_RANDOM) {
+                if (! isset($options['random-order-seed'])) {
+                    $options['random-order-seed'] = time();
```

Comment:
`time` was preferred over `random_int` to be compatible with native PHPUnit behavior